server/src/metrics/client_impact_metrics.rs

```diff
@@ -113,8 +113,24 @@ impl MetricsCache {
                         .insert(reduced_metric.impact_metric.name.clone(), reduced_metric);
                 }
             }
-            self.impact_metrics
-                .insert(key, aggregated_metrics.into_values().collect());
+            let layered_metrics = aggregated_metrics
+                .into_values()
+                .map(|mut metric| {
+                    for sample in &mut metric.impact_metric.samples {
+                        if let Some(labels) = &mut sample.labels {
+                            labels.insert("connected-via".to_string(), instance_id.clone());
```

Comment:
Not sure this is worth putting into a constant. Open to put it in one if anyone feels strongly about it though